index.js

```diff
@@ -8,19 +8,23 @@ const parser = require('./parser.js');
 const PLUGIN_NAME = 'gulp-closure-deps';
 
 const cache = {};
-let cwd, prefix, baseDir;
+let cwd, prefix;
 
-let extractDependency = function(filePath, contents) {
-  return parser.getDependencyCommand(contents, filePath, prefix);
-};
+let extractDependencyFactory = function(baseDir) {
+  return function(filePath, contents) {
+    return parser.getDependencyCommand(contents, filePath, prefix, baseDir);
+  };
+}
 
 module.exports = function(opt) {
   opt = opt || {};
   prefix = opt.prefix || '';
-  baseDir = opt.baseDir || '';
+  const baseDir = opt.baseDir || '';
```

Comment:
If the default value is '.', why not set it here so you do not need line#51 in parser.js.